UserTools/DirectionGridCheck/DirectionGridCheck.cpp

```diff
@@ -0,0 +1,245 @@
+
+    // MC trigger number
+    m_data->Stores.at("ANNIEEvent")->Get("MCTriggernum", fMCTriggerNum);
+
+    // ANNIE Event number
+    m_data->Stores.at("ANNIEEvent")->Get("EventNumber", fEventNumber);
+
+    auto get_flags = m_data->Stores.at("RecoEvent")->Get("EventFlagged", fEventStatusFlagged);
+
+    if ((fEventStatusFlagged) != 0) {
+        //  if (!fEventCutStatus){
+        Log("PhaseIITreeMaker Tool: Event was flagged with one of the active cuts.", v_debug, verbosity);
+        return true;
+    }
+
+    // Only check this event
+    //if (showEvent > 0 && (int)fEventNumber != showEvent) return true;
+    if (eventCount > 9)return true;
+
+    VertexGeometry* myvtxgeo = VertexGeometry::Instance();
+    Direction tempdir;
+
+
+    if (eventCount == 0) {
+        for (int l = 0; l < 100; l++) {
+            double phi = (6 * TMath::Pi() / 50) * l;
+            double theta = (TMath::Pi() / 200) * l;
+
+            while (phi > 2 * TMath::Pi()) {
+                std::cout << "correcting theta" << endl;
+                phi -= 2 * TMath::Pi();
+            }
+
+            tempdir.SetTheta(theta);
+            tempdir.SetPhi(phi);
+            AngularPlot->SetPoint(l, phi, theta);
+            std::cout << "Angular Plot set:" << l << ", " << phi << ", " << theta << endl;
+        }
+    }
+    RecoVertex* truevtx = 0;
+    RecoVertex* recovtx = 0;
+    auto get_muonMC = m_data->Stores.at("RecoEvent")->Get("TrueVertex", truevtx);
+    auto get_muonReco = m_data->Stores.at("RecoEvent")->Get("ExtendedVertex", recovtx);
+    double truePhi = truevtx->GetDirection().GetPhi();
+    double trueTheta = truevtx->GetDirection().GetTheta();
+    double RecoDirX = recovtx->GetDirection().X();
+    double RecoDirY = recovtx->GetDirection().Y();
+    double RecoDirZ = recovtx->GetDirection().Z();
+    double recoPhi = atan(RecoDirY/RecoDirX);
+    double recoTheta = asin(RecoDirX * RecoDirX + RecoDirY * RecoDirY);
+    if (trueTheta < 0) trueTheta += 2*TMath::Pi();
+    if (trueTheta > 2 * TMath::Pi()) trueTheta -= 2 * TMath::Pi();
+    TrueDirPlot->SetPoint(eventCount, truePhi, trueTheta);
+    std::cout << "True Plot set: " << eventCount << ", " << truePhi << ", " << trueTheta << endl;
+    Direction MRDDirection = this->findDirectionMRD();
+    MRDDirPlot->SetPoint(eventCount, MRDDirection.GetPhi(), MRDDirection.GetTheta());
+    std::cout << "MRD Plot set: " << eventCount << ", " << MRDDirection.GetPhi() << ", " << MRDDirection.GetTheta() << endl;
+    RecoDirPlot->SetPoint(eventCount, recoPhi, recoTheta);
+    std::cout << "recoPlot set: " << eventCount << ", " << recoPhi << ", " << recoTheta << endl;
+
+    eventCount++;
+  return true;
+}
+
+
+bool DirectionGridCheck::Finalise(){
+    fOutput_tfile->cd();
+    AngularPlot->Write("Angular Grid");
+    TrueDirPlot->Write("trueDirection");
+    MRDDirPlot->Write("MRDDirection");
+    RecoDirPlot->Write("recoDirection");
+    fOutput_tfile->Write();
+    fOutput_tfile->Close();
```

Comment:
you should be able to `delete` the TFile after closing it, as well as the TGraphs